src/scmrepo/git/__init__.py

```diff
@@ -11,7 +11,6 @@
 from typing import TYPE_CHECKING, Callable, Dict, Iterable, Optional, Tuple, Type, Union
 
 from funcy import cached_property, first
-from git import GitCommandNotFound
```

Comment:
gitpython raises an `ImportError` the first time it tries to find CLI git and it is unavailable, `GitCommandNotFound` is only raised on subsequent attempts to find CLI git. gitpython import should be lazy, otherwise `import scmrepo.git` will fail if CLI git is unavailable

Both the `ImportError` and `GitCommandNotFound` should be handled within the gitpython backend and not at this level